src/Oriflame.RedisMessaging.ReliableDelivery/Subscribe/MessageHandler.cs

```diff
@@ -35,18 +38,42 @@ public void HandleMessage(Message message)
             lock (_lock)
             {
                 HandleMessageImpl(message);
+                LastActivityAt = Now;
             }
         }
 
-        public void HandleNewestMessages()
+        protected virtual DateTime Now => DateTime.Now;

```

Comment:
Considrer using UtcNow. BTW what is the idea behind having virtual "Now"?

Reply:
Changed to `DateTime.UtcNow`. Property `MessageHandler.Now` is here to wrap usage of "now" moment across a class. It is `virtual` so that consumers have a way how to modulate time based on their needs, e.g. they time can be frozen or it can tick with different rate.